pandas/tests/reshape/concat/test_append.py

```diff
@@ -198,8 +196,12 @@ def test_append_same_columns_type(self, index):
         ser = Series([7, 8], index=ser_index, name=2)
         result = df.append(ser)
         expected = DataFrame(
-            [[1.0, 2.0, 3.0], [4, 5, 6], [7, 8, np.nan]], index=[0, 1, 2], columns=index
+            [[1, 2, 3.0], [4, 5, 6], [7, 8, np.nan]], index=[0, 1, 2], columns=index
         )
+        # integer dtype is preserved for columns present in ser.index
```

Comment:
assert_frame_equal compares these dtypes no? (i guess ok that you are being explicit as well)